### PR TITLE
Fixed looking up inlined action in the implementations objects

### PR DESCRIPTION
**Pull request by Andarist**

### CODE REVIEW

File: packages/core/src/actions.ts [hunk 1]
```diff
@@ -50,9 +53,13 @@ export function getActionFunction<
 
 export function resolveActionObject(
   actionObject: BaseActionObject,
-  actionFunctionMap?: ActionFunctionMap<any, any>
+  actionFunctionMap: ActionFunctionMap<any, any>
 ): BaseActionObject {
+  if (isDynamicAction(actionObject) || isExecutableAction(actionObject)) {
+    return actionObject;
+  }
```
Comment (Andarist) [inline packages/core/src/actions.ts:60]: This is the core of the fix

File: packages/core/test/actions.test.ts [hunk 2]
```diff
@@ -415,6 +415,62 @@ describe('entry/exit actions', () => {
...
+      );
+
+      const actor = interpret(machine).start();
+      actor.send({ type: 'EV' });
+
+      expect(spy).not.toHaveBeenCalled();
+      expect(actor.getSnapshot().context.assigned).toBe(true);
+    });
+
+    it("shouldn't use referenced custom action over a builtin one when there is a naming conflict", () => {
```
Comment (davidkpiano) [inline packages/core/test/actions.test.ts:445]:
```suggestion
    it("shouldn't use inline named custom action over a builtin one when there is a naming conflict", () => {
```

**Review by davidkpiano — COMMENTED** [review]
This changes the behavior from v4, right? That inline named actions will _not_ be replaced with equally named referenced actions?

I'm okay with the change, just want to double check (and we should changeset this)

File: packages/core/test/actions.test.ts (hunk 2 above)
Comment (Andarist) [reply packages/core/test/actions.test.ts:445]: ah, copy-pasta from the other added test case went wrong 😅

File: packages/core/test/actions.test.ts [hunk 3]
```diff
@@ -415,6 +415,62 @@ describe('entry/exit actions', () => {
       expect(actual).toEqual(['loaded entry']);
     });
 
+    it("shouldn't use referenced custom action over a builtin one when there is a naming conflict", () => {
```
Comment (Andarist) [inline packages/core/test/actions.test.ts:418]:
```suggestion
    it("shouldn't use a referenced custom action over a builtin one when there is a naming conflict", () => {
```

**Comment by Andarist** [comment]
> This changes the behavior from v4, right? That inline named actions will not be replaced with equally named referenced actions?

Yeah. I would consider this to be a plain bug fix though. I doubt somebody has been intentionally relying on this one.

**Review by davidkpiano — APPROVED** [review]
TODO changeset:

> Named inline action functions will not be replaced by equally named referenced actions.

**Comment by Andarist** [comment]
@davidkpiano changeset added

**Review by davidkpiano — APPROVED** [review]

**Pull request merged**